qa/L0_metrics/histogram_metrics_test.py

```diff
@@ -0,0 +1,189 @@
+            time_out = 10
+            while (len(user_data) == 0) and time_out > 0:
+                time_out = time_out - 1
+                time.sleep(1)
+
+            # Validate the results
+            for i in range(len(user_data)):
+                # Check for the errors
+                self.assertNotIsInstance(
+                    user_data[i], InferenceServerException, user_data[i]
+                )
+
+    def test_ensemble_decoupled(self):
+        wait_secs = 0.1
+        responses_per_req = 3
+        total_reqs = 3
+
+        # Infer
+        inputs = []
+        outputs = []
+        inputs.append(grpcclient.InferInput("INPUT0", [1], "FP32"))
+        inputs.append(grpcclient.InferInput("INPUT1", [1], "UINT8"))
+        outputs.append(grpcclient.InferRequestedOutput("OUTPUT"))
+
+        # Create the data for the input tensor.
+        input_data_0 = np.array([wait_secs], np.float32)
+        input_data_1 = np.array([responses_per_req], np.uint8)
+
+        # Initialize the data
+        inputs[0].set_data_from_numpy(input_data_0)
+        inputs[1].set_data_from_numpy(input_data_1)
+
+        # Send requests to ensemble decoupled model
+        for request_num in range(1, total_reqs + 1):
+            ensemble_model_name = "ensemble"
+            decoupled_model_name = "async_execute_decouple"
+            non_decoupled_model_name = "async_execute"
+            self.async_stream_infer(ensemble_model_name, inputs, outputs)
+
+            # Checks metrics output
+            first_response_family = "nv_inference_first_response_histogram_ms"
+            histogram_dict = self.get_histogram_metrics(first_response_family)
+
+            # Test ensemble model metrics
+            ensemble_model_count = get_histogram_metric_key(
+                first_response_family, ensemble_model_name, "1", "count"
+            )
+            ensemble_model_sum = get_histogram_metric_key(
+                first_response_family, ensemble_model_name, "1", "sum"
+            )
+            self.assertIn(ensemble_model_count, histogram_dict)
+            self.assertEqual(histogram_dict[ensemble_model_count], request_num)
+            self.assertIn(ensemble_model_sum, histogram_dict)
+            self.assertGreaterEqual(
+                histogram_dict[ensemble_model_sum],
+                2 * wait_secs * MILLIS_PER_SEC * request_num,
+            )
+
+            # Test decoupled model metrics
+            decoupled_model_count = get_histogram_metric_key(
+                first_response_family, decoupled_model_name, "1", "count"
+            )
+            decoupled_model_sum = get_histogram_metric_key(
+                first_response_family, decoupled_model_name, "1", "sum"
+            )
+            self.assertIn(decoupled_model_count, histogram_dict)
+            self.assertEqual(histogram_dict[decoupled_model_count], request_num)
+            self.assertIn(decoupled_model_sum, histogram_dict)
+            self.assertGreaterEqual(
```

Comment:
Can you also assert the number is within a range?

Reply:
Updated